swap/src/protocol/bob.rs

```diff
@@ -154,31 +150,31 @@ impl SwapFactory {
             self.alice_address.clone(),
         )
     }
-}
 
-async fn init_bob_state(
-    btc_to_swap: bitcoin::Amount,
-    xmr_to_swap: monero::Amount,
-    bob_btc_wallet: Arc<bitcoin::Wallet>,
-    config: Config,
-) -> Result<BobState> {
-    let amounts = SwapAmounts {
-        btc: btc_to_swap,
-        xmr: xmr_to_swap,
-    };
-
-    let refund_address = bob_btc_wallet.new_address().await?;
-    let state0 = bob::State0::new(
-        &mut OsRng,
-        btc_to_swap,
-        xmr_to_swap,
-        config.bitcoin_cancel_timelock,
-        config.bitcoin_punish_timelock,
-        refund_address,
-        config.monero_finality_confirmations,
-    );
-
-    Ok(BobState::Started { state0, amounts })
+    async fn init_state(
```

Comment:
I would call this `initial_state` (or maybe even `make_initial_state`) because that is what it returns. If something is called `init_state`, I would expect it to take `&mut self` so it can actually initialize something on `self`.